flink-table/flink-sql-gateway/src/main/java/org/apache/flink/table/gateway/rest/header/util/SQLGatewayUrlPrefixDecorator.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.gateway.rest.header.util;
+
+import org.apache.flink.client.program.rest.UrlPrefixDecorator;
+import org.apache.flink.runtime.rest.messages.MessageHeaders;
+import org.apache.flink.runtime.rest.messages.MessageParameters;
+import org.apache.flink.runtime.rest.messages.RequestBody;
+import org.apache.flink.runtime.rest.messages.ResponseBody;
+import org.apache.flink.runtime.rest.versioning.RestAPIVersion;
+import org.apache.flink.table.gateway.rest.util.SqlGatewayRestAPIVersion;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.stream.Collectors;
+
+/**
+ * This class decorates the URL of the original message headers by adding a prefix. The purpose of
+ * this decorator is to provide a consistent URL prefix for all the REST API endpoints while
+ * preserving the behavior of the original message headers.
+ *
+ * @param <R> the type of the request body
+ * @param <P> the type of the response body
+ * @param <M> the type of the message parameters
+ */
+public class SQLGatewayUrlPrefixDecorator<
+                R extends RequestBody, P extends ResponseBody, M extends MessageParameters>
+        extends UrlPrefixDecorator {
```

Comment:
Why is `SQLGatewayUrlPrefixDecorator` required? There is already UrlPrefixDecorator which this class is using as a super class. I don't understand why we need a new class. We can simply use UrlPrefixDecorator .

Reply:
The overridden method `getSupportedAPIVersions()` of SQLGatewayUrlPrefixDecorator will return SQL Gateway's stable REST API versions. UrlPrefixDecorator will be generic and can be used on the PyFlink side, which will need to fetch different stable rest API versions.

Reply:
The overridden method `getSupportedAPIVersions()` of SQLGatewayUrlPrefixDecorator will return SQL Gateway's stable REST API versions. UrlPrefixDecorator will be generic and can be used on the PyFlink side, which will need to fetch RuntimeRestAPIVersion stable versions.

Reply:
@tweise @mxm , I can get rid off `MonitoringApiMessageHeaders`, and make `getSupportedAPIVersions` a public method of `UrlPrefixDecorator`. Please let me know if you want me to do so. Thanks.

Reply:
+1, that would probably make it a bit easier to understand

Reply:
Thanks, updated correspondingly.

Reply:
Thanks!